localstack/services/ec2/resource_providers/aws_ec2_instance.py

```diff
@@ -303,4 +303,31 @@ def update(
 
 
         """
-        raise NotImplementedError
+        desired_state = request.desired_state
+        ec2 = request.aws_client_factory.ec2
+
+        groups = desired_state.get("SecurityGroups", desired_state.get("SecurityGroupIds"))
```

Comment:
That seems like a remnant from the way the old provider worked (fetch_state), but guess we can allow it for now